modules/apps/commerce/commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CPDefinitionsImporter.java

```diff
@@ -902,8 +894,8 @@ private CPInstance _importCPInstance(
 		}
 
 		CPInstance cpInstance = _cpInstanceLocalService.addCPInstance(
-			externalReferenceCode, cpDefinitionId, cpDefinition.getGroupId(),
-			sku, null, manufacturerPartNumber, true,
+			StringPool.BLANK, cpDefinitionId, cpDefinition.getGroupId(), sku,
```

Comment:
hey @amosfong I think this is not the solution for the problem, we need to find a new way for generating SKU code not get rid of them

Reply:
@amosfong you should instead use “sku+id” for externalReferenceCode.

Reply:
I didn't see the externalReferenceCode used anywhere so I didn't think it was important.

> @amosfong you should instead use “sku+id” for externalReferenceCode.

do you mean sku+companyId?

Reply:
Hi @amosfong I think @ivicac meant sku + cpDefinitionId

Reply:
thx @andrea-ale-sbarra  :)

Reply:
@amosfong cpDefinitionId and cpInstanceId are always unique, just to let you know

Reply:
> Hi @amosfong I think @ivicac meant sku + cpDefinitionId

ok, thanks. I will try this. Since cpDefinitionId is liferay generated, is it helpful to have this though?

Reply:
I don't think this will be enough. From the code here, using sku+cpDefinitionId would still create duplicate externalReferenceCodes:
https://github.com/liferay/liferay-portal/blob/41c58ed29f51d9c3ceafc608d1b678e3b2b67746/modules/apps/commerce/commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CPDefinitionsImporter.java#L528

Reply:
hey @amosfong you can add for this case sku+cpInstanceId to achieve uniqueness as cpInstanceId is unique.

Reply:
@ivicac ok I can do that

Reply:
Thanks for the help guys. I looked at it again and the ERC needs to be predictable so it can be looked up. I think this is proper solution: https://github.com/liferay-commerce/liferay-portal/pull/2516